encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/AesAlgorithm.java

```diff
@@ -47,51 +54,63 @@ private Cipher getUninitializedCipher() throws NoSuchAlgorithmException, NoSuchP
+                    NoSuchProviderException, IllegalBlockSizeException, InvalidAlgorithmParameterException,
+                    InvalidKeyException, InvalidKeySpecException, IOException {
+                Cipher cipher = getUninitializedCipher();
+                Key secretKeySpec = getSecretKeySpec(kwa);
 
-            byte[] ivBytes = new byte[IV_LENGTH];
+                byte[] ivBytes = new byte[IV_LENGTH];
 
-            SecureRandom secureRandom = new SecureRandom();
-            secureRandom.nextBytes(ivBytes);
+                SecureRandom secureRandom = new SecureRandom();
+                secureRandom.nextBytes(ivBytes);
 
-            GCMParameterSpec gcmSpecWithIv = new GCMParameterSpec(GCM_AUTH_TAG_LENGTH, ivBytes);
+                GCMParameterSpec gcmSpecWithIv = new GCMParameterSpec(GCM_AUTH_TAG_LENGTH, ivBytes);
 
-            cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec, gcmSpecWithIv);
-            byte[] encrypted = cipher.doFinal(plaintext.getBytes(charset));
+                cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec, gcmSpecWithIv);
+                byte[] encrypted = cipher.doFinal(plaintext.getBytes(charset));
 
-            // put together the iv and the encrypted bytes
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            outputStream.write(ivBytes);
-            outputStream.write(encrypted);
+                // put together the iv and the encrypted bytes
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                outputStream.write(ivBytes);
+                outputStream.write(encrypted);
+
+                String encryptedString =  Base64.getEncoder().encodeToString(outputStream.toByteArray());
+                return EncryptedValue.fromEncryptedString(encryptedString);
+            }
 
-            String encryptedString =  Base64.getEncoder().encodeToString(outputStream.toByteArray());
-            return EncryptedValue.fromEncryptedString(encryptedString);
         });
     }
 
     @Override
-    public String getDecryptedString(EncryptedValue encryptedValue, KeyWithAlgorithm kwa) {
+    public String getDecryptedString(final EncryptedValue encryptedValue, final KeyWithAlgorithm kwa) {
         checkArgument(kwa.getAlgorithm().equals(ALGORITHM_TYPE),
                 "key must be for AES algorithm but was %s", kwa.getAlgorithm());
 
-        return DecryptedStringSupplier.silently(() -> {
-            Cipher cipher = getUninitializedCipher();
-            Key secretKeySpec = getSecretKeySpec(kwa);
+        return DecryptedStringSupplier.silently(new DecryptedStringSupplier() {
+            @Override
+            public String get() throws BadPaddingException, IllegalBlockSizeException,
+                    InvalidAlgorithmParameterException, InvalidKeyException,
+                    InvalidKeySpecException, NoSuchAlgorithmException, NoSuchPaddingException,
```

Comment:
so, the silently gives you (in theory) a nice error, though I suppose it could do that anyway